### DISCUSSION TITLE
Getting colors in script

### DISCUSSION THREAD

**Discussion by mshamaseen** [unocss/unocss#1908] [Q&A] [answered]
I'm trying to get the colors defined in the unocss.config in the script in Vue.
Is there any function available in script like the theme in scss?

or maybe resolving the config like [tailwind resolve](https://tailwindcss.com/docs/configuration#referencing-in-java-script)

**Comment by zyyv** [comment]
Like this: 
```ts
import { createGenerator } from 'unocss'
import config from '../unocss.config'

const ctx = createGenerator(config)
console.log(ctx.config.theme)
```

**Reply by mshamaseen** [reply]
Thanks @zyyv , I wonder if this is documented anywhere so I can read further?

**Reply by zyyv** [reply]
> Thanks @zyyv , I wonder if this is documented anywhere so I can read further?

It goes beyond the simple use level, you may need to briefly look at its source code, or you can read our test code, it is very common in test code

**Reply by ackvf** [reply]
When I try to imporrt the config file to use with the createGenerator it gives me lots of build errors like this
```
✘ [ERROR] Could not resolve "fs"

    node_modules/@iconify/utils/lib/loader/node-loader.cjs:18:8:
      18 │ require('fs');
         ╵         ~~~~

  The package "fs" wasn't found on the file system but is built
  into node. Are you trying to bundle for node? You can use
  "--platform=node" to do that, which will remove this error.

✘ [ERROR] Could not resolve "process"

    node_modules/@antfu/install-pkg/dist/index.js:2:20:
      2 │ import process from "process";
        ╵                     ~~~~~~~~~

  The package "process" wasn't found on the file system but is
  built into node. Are you trying to bundle for node? You can
  use "--platform=node" to do that, which will remove this
  error.
```

Using `--platform=node` doesn't fix it and is also wrong anyway.

**Comment by mshamaseen** [comment]
@zyyv I'm getting "Uncaught TypeError: The "original" argument must be of type Function"
![image](https://user-images.githubusercontent.com/38951661/204291725-809f5a73-586c-4f41-b55c-f74572bac86a.png)

**Reply by mshamaseen** [reply]
ok I found something, in the `unocss,config.ts` I was using `import { defineConfig } from 'unocss\vite'` , when I changed it to `import { defineConfig } from 'unocss'` it worked. I don't know what is the issue here.

**Reply by zyyv** [reply]
https://github.com/unocss/unocss/blob/22c4e70d53870b9413399e0ed20a7fece610eaa3/packages/vite/src/index.ts#L23-L25
`import { defineConfig } from 'unocss\vite'` for `VitePluginConfig`

-------
https://github.com/unocss/unocss/blob/22c4e70d53870b9413399e0ed20a7fece610eaa3/packages/unocss/src/index.ts#L18-L20
This is the configuration we need

**Comment by mshamaseen** [comment, marked as answer]
this issue happens when trying to build while importing `import { createGenerator } from 'unocss'`

https://github.com/unocss/unocss/issues/1885

**Reply by zyyv** [reply]
Can you provide a minimal repo?

**Reply by mshamaseen** [reply]
@zyyv here you go:
https://codesandbox.io/s/vigilant-wu-dg43bc?file=/package.json

try to run `yarn run build`, you will get the error. but it is working fine on dev server..

**Reply by zyyv** [reply]
As https://github.com/unocss/unocss/issues/1885#issuecomment-1329973918 said.

And i would like to write a vite plugin to implement it.

**Reply by zyyv** [reply]
Hi @mshamaseen , Sorry to reply you so late, you can solve the problem like this:
```ts
// vite.config.ts
import { createGenerator } from https://github.com/unocss/unocss
import config from '../unocss.config'

const unoCtx = createGenerator(config)

export default defineConfig({
 // ...
  define:{
    // ...
    'import.meta.env.__UNO__': uno,
    'import.meta.env.__UNO_THEME__.': uno.config.theme
  }
})
```
Then, you can use `import.meta.env.__UNO__` or other define variants in `.vue`.

**Reply by mshamaseen** [reply]
@zyyv oh, using it as an env variable, good point.
thanks for your help, it is much appreciated.
